fuzzer/HttpResponse.cpp

```diff
@@ -7,7 +7,8 @@
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
 {
     http::Response response;
-    response.readData(reinterpret_cast<const char*>(data), size);
+    for (size_t i = 0; i < size; ++i)
+        response.readData(reinterpret_cast<const char*>(data), i);
```

Comment:
Is this just something speculative? libfuzzer already watches coverage and already tries to insert \0 chars at random positions in the input, sounds like you try to duplicate that here, but without checking if it improves coverage or not.

I.e. it seems this is not helpful, but perhaps I miss something.

Reply:
What I'm hoping to do here is to give the parser partial data. The fuzzer (as I understand, at least) generates data to trigger different paths/conditions in the code. One of the most important ones is the parser not having enough data to parse a token. Finding out all the different parser cases is really hard, so I'm trying to help the fuzzer by re-using a single data to cover all possible 'incomplete data' cases. 

I have to agree that this is probably useful for catching early issues and once we have a well-tested parser it probably wastes cycles. I'll consider reverting it.